### PR TITLE
Using generators in sabre/vobject where they make sense

### LINKED ISSUE #282: Use generators where we return arrays now
This is probably the last vobject 4 change. Even though we're already in beta, I think it's worth implementing.

### PR DESCRIPTION
**Pull request by evert**
Fixes #282

### CODE REVIEW

**Comment by evert** [comment]
Benchmarks

Before

```
__global__  |||||||||||||||||||||||||||||||||||||||||||||||||||| 57830ms, 100.0%
parse                                                              257ms,   0.4%
fb          |||||||||||||||||||||||||||||||||||||||||||||||||||| 57573ms,  99.6%

peak memory usage: 14M
current memory usage: 13M
```

After

```
__global__  |||||||||||||||||||||||||||||||||||||||||||||||||||| 57561ms, 100.0%
parse                                                              256ms,   0.4%
fb          |||||||||||||||||||||||||||||||||||||||||||||||||||| 57305ms,  99.6%

peak memory usage: 15M
current memory usage: 13M
```

Unfortunately this implies there's nearly no performance benefit, and slightly more memory usage. Because of this, I'll discard the patch.

File: lib/Cli.php [hunk 1]
```diff
@@ -583,7 +583,7 @@ protected function serializeComponent(Component $vObj) {
 
         };
 
-        $children = $vObj->children();
+        $children = iterator_to_array($vObj->children());
```
Comment (staabm) [inline lib/Cli.php]: I guess this was the culprit of not getting it faster... Did you try ArrayIterator::uksort?

Comment (staabm) [inline lib/Cli.php]: Same in the other places of iterator_to_array

Comment (evert) [inline lib/Cli.php]: In my benchmark I mostly tested things that use things like property-access. A lot of statements like this `$vcalendar->VEVENT->DTSTART` use (in this PR) multiple generators.

I was hoping there would be a fair bit of win there. Changing the iterator_to_array and ksort statements only really help with serializing, which is something I haven't really tested for. I suspect though that the win will be minimal too...

**Pull request closed without merging**